core/commands/refs.go

```diff
@@ -85,6 +86,16 @@ NOTE: List all references recursively by using the flag '-r'.
 			return
 		}
 
+		maxDepth, _, err := req.Option("max-depth").Int()
+		if err != nil {
+			res.SetError(err, cmdkit.ErrNormal)
+			return
+		}
+
+		if !recursive {
+			maxDepth = 1 // write only direct refs
```

Comment:
Should this be `0`? If I'm not mistaken, this patch will currently explore two levels when recursive is disabled. If that's the case, can we *also* write a sharness test?

Reply:
I see. This looks correct.